android/src/main/java/com/onegini/mobile/sdk/reactnative/network/AnonymousService.kt

```diff
@@ -1,17 +1,20 @@
 package com.onegini.mobile.sdk.reactnative.network
 
 import com.google.gson.JsonObject
-import com.onegini.mobile.sdk.reactnative.OneginiComponents
 import com.onegini.mobile.sdk.reactnative.model.ResourceRequestDetails
 import com.onegini.mobile.sdk.reactnative.network.client.ResourcesClient
+import com.onegini.mobile.sdk.reactnative.network.client.SecureResourceClient
 import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
 import io.reactivex.rxjava3.core.Single
 import io.reactivex.rxjava3.schedulers.Schedulers
+import javax.inject.Inject
+import javax.inject.Singleton
 
-class AnonymousService private constructor() {
-
-    private val applicationDetailsRetrofitClient: ResourcesClient = OneginiComponents.secureResourceClient.prepareSecuredAnonymousRetrofitClient(ResourcesClient::class.java)
-
+@Singleton
+class AnonymousService @Inject constructor(private val secureResourceClient: SecureResourceClient){
+    private val applicationDetailsRetrofitClient: ResourcesClient by lazy {
```

Comment:
correct retrofit client can also be injected. You can create new dagger module that will have provider methods and each can provide different retrofit client. Each provider would return the same type, thus you will need to use @Named annotation, so dagger knows which one to inject